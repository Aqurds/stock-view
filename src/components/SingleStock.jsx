import '../styles/single-stock.css';
import { useParams } from 'react-router-dom';
import { useEffect, useState } from 'react';
import axios from "axios";
import Header from './Header';
import { IoChevronBack } from 'react-icons/io5'

const API_KEY = '694e1015c577e6802d8f0e4f00ea92ef';
const API_BASE_URL = 'https://financialmodelingprep.com/api/v3/';
const INCOME_EXT = 'income-statement';


const SingleStock = () => {
  const stockName = useParams();
  // console.log(stockName)
  const [singleStockData, setSingleStockData] = useState(null);
  const [apiErrorMessage, setApiErrorMessage] = useState('');
  const [dataLoading, setDataLoading] = useState(true)

  useEffect(() => {
    async function fetchData() {
      try {
        const response = await axios.get(`${API_BASE_URL}${INCOME_EXT}/${stockName.id}?apikey=${API_KEY}`);
        console.log(response.data);
        setSingleStockData(response.data);
        setApiErrorMessage('');
        setDataLoading(false);
      } catch (error) {
        console.log(error.response.data['Error Message'])
        setApiErrorMessage(error.response.data['Error Message'])
        setDataLoading(false);
      }
    }
    fetchData();
  }, [])
  return (
    <div className="stock-details">
      <a href='/' className='go-back'>
        <IoChevronBack className='go-back-icon'/>
      </a>
      <Header />
      <h2 className='stock-header'>
        Stock name: <br/>
        <span className='stock-name'>{stockName.id}</span>
      </h2>
        { dataLoading && <p className='stock-loading'>Loading ...</p> }
        { apiErrorMessage && <p className='stock-loading'>{apiErrorMessage}<br/>Please go back and check another stock. Sorry for the inconvenience!</p> }
        { singleStockData && singleStockData.map((stockData) => {
          return(
            <div key={stockData.date} className='stock-data-by-date'>
              <p className='stock-date'>
                Reporting date: <br/>
                <span className='stock-date-speci'>{stockData.date}</span>
              </p>
              <div className='stock-details'>
                <div>
                  <p>Calender Year: {stockData.calendarYear}</p>
                  <p>Cost & Expenses: {stockData.costAndExpenses}</p>
                  <p>Cost of Revenue: {stockData.costOfRevenue}</p>
                  <p>Reported Currency: {stockData.reportedCurrency}</p>

                  <p>Eps: {stockData.eps}</p>
                  <p>Gross Profile: {stockData.grossProfit}</p>
                  <p>Net Income: {stockData.netIncome}</p>
                  <p>Final Link :  
                    <a className='final-link' target='_blank' href={stockData.finalLink}>
                      Link
                    </a>
                  </p>

                </div>
              </div>
            </div>
          )
        })}
    </div>
  )
};

export default SingleStock;