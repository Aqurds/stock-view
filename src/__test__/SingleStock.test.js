import React from 'react';
import { render } from '@testing-library/react';
import SingleStock from '../components/SingleStock';

describe('SingleStock component', () => {
  it('matches snapshot', () => {
    const { container } = render(<SingleStock />);
    expect(container).toMatchSnapshot();
  });
});
