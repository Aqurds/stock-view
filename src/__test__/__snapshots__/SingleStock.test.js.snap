// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleStock component matches snapshot 1`] = `
<div>
  <div
    class="stock-details"
  >
    <a
      class="go-back"
      href="/"
    >
      <svg
        class="go-back-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M328 112L184 256l144 144"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="48"
        />
      </svg>
    </a>
    <header
      class="header-wrapper"
      data-testid="header"
    >
      <svg
        class="stock-view-logo"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM464 96H345.94c-21.38 0-32.09 25.85-16.97 40.97l32.4 32.4L288 242.75l-73.37-73.37c-12.5-12.5-32.76-12.5-45.25 0l-68.69 68.69c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0L192 237.25l73.37 73.37c12.5 12.5 32.76 12.5 45.25 0l96-96 32.4 32.4c15.12 15.12 40.97 4.41 40.97-16.97V112c.01-8.84-7.15-16-15.99-16z"
        />
      </svg>
      <h1
        class="header-text"
      >
        Stock View
      </h1>
      <div
        class="header-icon-wrapper"
      >
        <svg
          class="header-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
          />
          <path
            d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
          />
        </svg>
        <svg
          class="header-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 176a80 80 0 1080 80 80.24 80.24 0 00-80-80zm172.72 80a165.53 165.53 0 01-1.64 22.34l48.69 38.12a11.59 11.59 0 012.63 14.78l-46.06 79.52a11.64 11.64 0 01-14.14 4.93l-57.25-23a176.56 176.56 0 01-38.82 22.67l-8.56 60.78a11.93 11.93 0 01-11.51 9.86h-92.12a12 12 0 01-11.51-9.53l-8.56-60.78A169.3 169.3 0 01151.05 393L93.8 416a11.64 11.64 0 01-14.14-4.92L33.6 331.57a11.59 11.59 0 012.63-14.78l48.69-38.12A174.58 174.58 0 0183.28 256a165.53 165.53 0 011.64-22.34l-48.69-38.12a11.59 11.59 0 01-2.63-14.78l46.06-79.52a11.64 11.64 0 0114.14-4.93l57.25 23a176.56 176.56 0 0138.82-22.67l8.56-60.78A11.93 11.93 0 01209.94 26h92.12a12 12 0 0111.51 9.53l8.56 60.78A169.3 169.3 0 01361 119l57.2-23a11.64 11.64 0 0114.14 4.92l46.06 79.52a11.59 11.59 0 01-2.63 14.78l-48.69 38.12a174.58 174.58 0 011.64 22.66z"
          />
        </svg>
      </div>
    </header>
    <h2
      class="stock-header"
    >
      Stock name: 
      <br />
      <span
        class="stock-name"
      />
    </h2>
    <p
      class="stock-loading"
    >
      Loading ...
    </p>
  </div>
</div>
`;
